engine/src/org/pentaho/di/trans/steps/httppost/HTTPPOST.java

```diff
@@ -494,7 +495,8 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
       data.realProxyHost = environmentSubstitute( meta.getProxyHost() );
       data.realProxyPort = Const.toInt( environmentSubstitute( meta.getProxyPort() ), 8080 );
       data.realHttpLogin = environmentSubstitute( meta.getHttpLogin() );
-      data.realHttpPassword = environmentSubstitute( meta.getHttpPassword() );
+      data.realHttpPassword =
+          Encr.decryptPasswordOptionallyEncrypted( environmentSubstitute( meta.getHttpPassword() ) );
```

Comment:
This is what I expected everything else to look like - see my comment on the PDI-15652. So if it's OK to do it simply and inline like this here, why can't everything else be this simple? so why the resolvePassword method added to stuff above, but this one is OK to be a single statement? I prefer the single statement.